### ISSUE TITLE
optionally align items in coc-list

### ISSUE THREAD

**Issue by elkowar** [neoclide/coc.nvim#2630]
**Is your feature request related to a problem? Please describe.**
 currently, the coc-list window looks rather busy, and is visually hard to parse.

**Describe the solution you'd like**
 there should be an option to align the different columns in the lists, specifically things like the diagnostics list.

**Describe alternatives you've considered**
If possible, it would also be nice if it was possible to hide certain parts of the diagnostics output

**Additional context**
If you'd consider merging a PR, I'd be more than happy to try and implement this!

**Comment by druskus20** [comment]
This has been bothering me in the past. It would be a nice addition though.

**Comment by chemzqm** [comment]
Can't understand, what do you mean by `busy` and `align items`?

**Comment by elkowar** [comment]
This is how it looks now:
![coc-now](https://user-images.githubusercontent.com/5300871/99878763-e13a0000-2c07-11eb-97e9-c1d7dde2136a.png)

If the content was aligned in columns,it would be easier to read, as seen here:
![coc-better](https://user-images.githubusercontent.com/5300871/99878767-e72fe100-2c07-11eb-921e-7702935f8724.png)

The _best_ case, In my opinion, would be if it was possible to configure the format of the list with more flexibility, although i'm not quite sure how viable that would be to implement. Specifically hiding the `[rustc E0308]` error id would likely not be easy, whereas moving, hiding or formatting the file path to the diagnostic should be more than doable. This could look something like this:
![coc-more-better](https://user-images.githubusercontent.com/5300871/99878805-3f66e300-2c08-11eb-890c-e205dc77a296.png)

**Comment by buffet** [comment]
This is actually something that I've thought about implementing myself a while ago, would be epic to finally have this!
